### PR TITLE
Refactor PoliciesListService into a instantiable class

### PR DESCRIPTION
**Pull request by guicassolato**

### CODE REVIEW

File: app/services/policies/policies_list_service.rb [hunk 1]
```diff
@@ -10,26 +10,49 @@ class Policies::PoliciesListService
...
+
+  def initialize(account, builtin: true)
+    @account = account
+    @builtin = builtin
+  end
+
+  attr_reader :account
+
+  def builtin?
+    !!@builtin
```
Comment (Martouta) [inline app/services/policies/policies_list_service.rb]: Another option is in the line 25 doing `@builtin = !!builtin` and here just `@builtin` so we don't have to do convert through `!!` for every request to `builtin?` 😄

Comment (guicassolato) [reply app/services/policies/policies_list_service.rb]: In fact, we don't even need to convert it if we don't want to. `builtin?` is only meant for readability. This works just the same:

```ruby
def builtin?
  @builtin
end

Or even this:

```ruby
attr_reader :builtin
alias builtin? builtin
```

Comment (Martouta) [reply app/services/policies/policies_list_service.rb]: Yup, also good 👍 😎

Comment (josemigallas) [reply app/services/policies/policies_list_service.rb]: I like the latter the most

File: app/services/policies/policies_list_service.rb [hunk 2]
```diff
@@ -10,26 +10,49 @@ class Policies::PoliciesListService
...
 
-  def self.raise_policies_list_error(error)
+  def raise_policies_list_error(error)
     raise PoliciesListServiceError, I18n.t('errors.messages.apicast_not_found', url: apicast_registry_url, error: error)
   end
 
-  def self.policies_from_account(account)
+  def policies_from_account
     return unless account.provider_can_use?(:policy_registry)
     PolicyList.new(account.policies)
```
Comment (josemigallas) [inline app/services/policies/policies_list_service.rb]: No need for `@account` here?

File: app/services/policies/policies_list_service.rb (hunk 1 above)
Comment (guicassolato) [reply app/services/policies/policies_list_service.rb]: Done

File: app/services/policies/policies_list_service.rb (hunk 2 above)
Comment (Martouta) [reply app/services/policies/policies_list_service.rb]: @josemigallas `account` is calling the method `account`, which exists for the `attr_reader :account` which is the same as:
```ruby
def account
  @account
end
```

File: app/services/policies/policies_list_service.rb [hunk 3]
```diff
@@ -10,26 +10,46 @@ class Policies::PoliciesListService
...
   class PoliciesListServiceError < RuntimeError; end
 
-  def self.call(account, builtin: true)
+  def self.apicast_registry_url
+    ThreeScale.config.sandbox_proxy.apicast_registry_url
+  end
+
+  delegate :apicast_registry_url, to: 'self.class'
+
+  def self.call(*args)
```
Comment (josemigallas) [inline app/services/policies/policies_list_service.rb]: Can one actually pass whatever in args?

File: app/services/policies/policies_list_service.rb (hunk 2 above)
Comment (guicassolato) [reply app/services/policies/policies_list_service.rb]: Nope. It's here already: https://github.com/3scale/porta/blob/5496b80bdec77587ae325fff4dd3591fc706e0f9/app/services/policies/policies_list_service.rb#L28

File: app/services/policies/policies_list_service.rb (hunk 3 above)
Comment (guicassolato) [reply app/services/policies/policies_list_service.rb]: Yep. That means `#call` requires no argument. It's all forwarded to the initializer.

Comment (Martouta) [reply app/services/policies/policies_list_service.rb]: @josemigallas You can pass as many arguments as you want, this is called **splat operators** 😄

Comment (Martouta) [reply app/services/policies/policies_list_service.rb]: @josemigallas You can check it in https://medium.freecodecamp.org/rubys-splat-and-double-splat-operators-ceb753329a78 👍

Comment (josemigallas) [reply app/services/policies/policies_list_service.rb]: What sense does this make? The constructor requires account and perhaps builtin, why the static method does not have the same signature? I could throw a confusing error, no?

Comment (Martouta) [reply app/services/policies/policies_list_service.rb]: @josemigallas it behaves the same way as if you call to the initialize directly with the wrong params. It is just more dynamic and delegates to it so if you change it there, you don't have to change it to many places. In Java for example (at least in old versions), you can't just delegate 'whatever you receive' and if you change the method arguments, you have to change the whole change of calls, even if that involves many files, here you just need in that same place and that's it.

Comment (josemigallas) [reply app/services/policies/policies_list_service.rb]: OK if it's the ruby way

**Pull request merged**